android/src/main/java/com/polidea/flutter_ble_lib/delegate/DiscoveryDelegate.java

```diff
@@ -0,0 +1,192 @@
+                    @Override
+                    public void onError(BleError error) {
+                        resolver.onError(error);
+                    }
+                });
+    }
+
+    private void characteristics(String deviceId, final String serviceUuid, final MethodChannel.Result result) {
+        adapter.getServicesForDevice(deviceId,
+                new OnSuccessCallback<Service[]>() {
+                    @Override
+                    public void onSuccess(final Service[] data) {
+                        for (final Service service : data) {
+                            if (service.getUuid().equals(UUID.fromString(serviceUuid))) {
+                                adapter.getCharacteristicsForService(service.getId(),
+                                        new OnSuccessCallback<Characteristic[]>() {
+                                            @Override
+                                            public void onSuccess(Characteristic[] data) {
+                                                CharacteristicsResponse characteristicsResponse = new CharacteristicsResponse(data, service);
+                                                try {
+                                                    String json = new CharacteristicsResponseConverter().toJson(characteristicsResponse);
+                                                    result.success(json);
+                                                } catch (JSONException e) {
+                                                    result.error(e.toString(), e.getMessage(), null);
+                                                }
+                                            }
+                                        },
+                                        new OnErrorCallback() {
+                                            @Override
+                                            public void onError(BleError error) {
+                                                result.error(String.valueOf(error.errorCode.code), error.reason, null);
+                                            }
+                                        });
+                                return;
+                            }
+                        }
+
+                    }
+                }, new OnErrorCallback() {
+                    @Override
+                    public void onError(BleError error) {
+                        result.error(String.valueOf(error.errorCode.code), error.reason, null);
+                    }
+                });
+    }
+
+    private void services(String deviceId, final MethodChannel.Result result) {
+        adapter.getServicesForDevice(deviceId,
+                new OnSuccessCallback<Service[]>() {
+                    @Override
+                    public void onSuccess(final Service[] data) {
+                        ServiceConverter converter = new ServiceConverter();
+                        JSONArray jsonArray = new JSONArray();
```

Comment:
Because, as the name implies, it should convert a single Service. We could create another converter, that would take care of a list of services, I guess?